server/cloudformation/apps.go

```diff
@@ -0,0 +1,88 @@
+package cloudformation
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/jinzhu/gorm"
+	"github.com/remind101/empire"
+)
+
+// appClient mocks the Empire interface we use.
+type appClient interface {
+	AppsFind(empire.AppsQuery) (*empire.App, error)
+	Create(context.Context, empire.CreateOpts) (*empire.App, error)
+	Destroy(context.Context, empire.DestroyOpts) error
+}
+
+// AppProperties represents the properties for the Custom::EmpireApp
+type AppProperties struct {
+	Name *string
+}
+
+// AppResource is a Provisioner that will manage an Empire application
+type AppResource struct {
+	empire appClient
+}
+
+func (p *AppResource) Provision(req Request) (id string, data interface{}, err error) {
+	ctx := context.Background()
+	user := NewUser()
+
+	properties := req.ResourceProperties.(*AppProperties)
+
+	switch req.RequestType {
+	case Create:
+		name := properties.Name
+		app, err := p.empire.AppsFind(empire.AppsQuery{
+			Name: name,
+		})
+		if err != nil && err != gorm.RecordNotFound {
+			return "", nil, err
+		}
+
+		app, err = p.empire.Create(ctx, empire.CreateOpts{
+			User:    user,
+			Name:    *name,
+			Message: "Creating app via Cloudformation",
+		})
+		if err != nil {
+			return "", nil, err
+		}
+
+		return app.ID, map[string]string{"Id": app.ID}, nil
```

Comment:
My original assumption was that we couldn't use `Ref` with the physical Id, so I added the Id as a returned property that we could use with `GetAtt`, but I don't think we need to do that. Can just `return app.ID, nil, nil` here.